grpc/src/main/java/com/linecorp/armeria/server/grpc/GrpcServiceBuilder.java

```diff
@@ -567,22 +569,31 @@ public GrpcServiceBuilder useClientTimeoutHeader(boolean useClientTimeoutHeader)
     }
 
     /**
-     * Adds a {@link GrpcHealthCheckService} to this {@link GrpcServiceBuilder}.
+     * Sets a {@link GrpcHealthCheckService} to this {@link GrpcServiceBuilder}.
+     *
+     * @see <a href="https://github.com/grpc/grpc/blob/master/doc/health-checking.md">GRPC Health Checking Protocol</a>
      */
-    public GrpcServiceBuilder addGrpcHealthCheckService(GrpcHealthCheckService grpcHealthCheckService) {
+    public GrpcServiceBuilder grpcHealthCheckService(GrpcHealthCheckService grpcHealthCheckService) {
         requireNonNull(grpcHealthCheckService, "grpcHealthCheckService");
+        if (enableDefaultHealthCheckService) {
+            throw new IllegalStateException("default gRPC health check service is enabled already.");
+        }
         this.grpcHealthCheckService = grpcHealthCheckService;
         return this;
     }
 
     /**
-     * Sets a {@link GrpcHealthCheckService} to this {@link GrpcServiceBuilder}.
+     * Sets a default {@link GrpcHealthCheckService} to this {@link GrpcServiceBuilder}.
      * The gPRC health check service manages only the health checker that determines
      * the healthiness of the {@link Server}.
+     *
+     * @see <a href="https://github.com/grpc/grpc/blob/master/doc/health-checking.md">GRPC Health Checking Protocol</a>
      */
-    public GrpcServiceBuilder enableGrpcHealthCheckService() {
-        grpcHealthCheckService = GrpcHealthCheckService.builder()
-                                                       .build();
+    public GrpcServiceBuilder enableDefaultGrpcHealthCheckService(boolean enableDefaultHealthCheckService) {
```

Comment:
I prefer not to have `Default` keyword in the method name for simplicity's sake.